### DISCUSSION TITLE
Adaptive Mesh Refinement Feature

### DISCUSSION THREAD

**Discussion by novtov354** [techMathGroup/openHFDIB-DEM#14] [Q&A]
Dear developer,

Thank you for sharing the code,

I found that the Adaptive Mesh Refinement (AMR) feature can no longer be used in the new version, although it is described in README.md that this version enables running with AMR.
However, all new version tutorials are provided with staticMeshFv, and when I tried to activate dynamicRefineFvMesh I always got errors.

Could you please inform me, what is the current status of the AMR in the solver?

Your responses are very much appreciated

Thank you

Best Regards

Novan

**Comment by MartinIsoz** [comment]
Dear Novan,

yes, it is possible that the adaptive mesh refinement is broken at the moment. Unfortunately, as we are a small research team, some features may be temporarily put on hold while we focus on some other part of the code. The reason behind the broken AMR is probably the changed paralleziation.

At the moment, we do not have a sufficient manpower to treat the problem at once. Thus, I see two possible ways of going forward from this point. 

1. If you will be kind enough to provide us with a minimal working example leading to the solver crushing, we may try to solve the issue once we have some spare time.
2. If you will the solve the issue by yourself and create a pull request, we will check that the performance of the solution and include it in the upstream solver version with authorship credits to you.

Best regards,
Martin
